messagingpush/src/main/java/io/customer/messagingpush/CustomerIOPushNotificationHandler.kt

```diff
@@ -0,0 +1,192 @@
+        val flags = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            PendingIntent.FLAG_IMMUTABLE or PendingIntent.FLAG_UPDATE_CURRENT
+        } else {
+            PendingIntent.FLAG_UPDATE_CURRENT
+        }
+
+        val icon = context.applicationInfo.icon
+
+        // set title and body
+        val title = bundle.getString(TITLE_KEY) ?: remoteMessage.notification?.title ?: ""
+        val body = bundle.getString(BODY_KEY) ?: remoteMessage.notification?.body ?: ""
+
+        val channelId = context.packageName
+        val defaultSoundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)
+        val notificationBuilder = NotificationCompat.Builder(context, channelId)
+            .setSmallIcon(icon)
+            .setContentTitle(title)
+            .setContentText(body)
+            .setAutoCancel(true)
+            .setSound(defaultSoundUri)
+            .setTicker(applicationName)
+
+        try {
+            // check for image
+            val notificationImage = bundle.getString(IMAGE_KEY)
+            if (notificationImage != null) {
+                addImage(notificationImage, notificationBuilder, body)
+            }
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
+
+        val notificationManager =
+            context.getSystemService(FirebaseMessagingService.NOTIFICATION_SERVICE) as NotificationManager
+
+        // Since android Oreo notification channel is needed.
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val channel = NotificationChannel(
+                channelId,
+                CHANNEL_NAME,
+                NotificationManager.IMPORTANCE_DEFAULT
+            )
+            notificationManager.createNotificationChannel(channel)
+        }
+
+        // set pending intent
+        val pushContentIntent = Intent(CustomerIOPushReceiver.ACTION)
+        pushContentIntent.setClass(context, CustomerIOPushReceiver::class.java)
+
+        pushContentIntent.putExtras(bundle)
+        val notificationClickedIntent = PendingIntent.getBroadcast(
+            context,
+            requestCode,
+            pushContentIntent,
+            flags
+        )
+        notificationBuilder.setContentIntent(notificationClickedIntent)
+
+        val notification = notificationBuilder.build()
+        notificationManager.notify(requestCode, notification)
+    }
+
+
+    private fun addImage(
+        imageUrl: String,
+        builder: NotificationCompat.Builder,
+        body: String
+    ) = runBlocking {
+        val style = NotificationCompat.BigPictureStyle()
+            .bigLargeIcon(null)
+            .setSummaryText(body)
+        val url = URL(imageUrl)
+        withContext(Dispatchers.IO) {
+            try {
+                val input = url.openStream()
+                BitmapFactory.decodeStream(input)
+            } catch (e: IOException) {
+                null
+            }
```

Comment:
Are there more exceptions that can be thrown besides `IOException` from this `try{}` block? I agree that this is a good approach to try to create a Bitmap and if for any reason it fails, just continue on with the push notification and display it. My concern is that there could be a different exception thrown and the SDK code does not catch it resulting in a runtime crash.

Reply:
Good point, it would be safer to go with the `Exception` class and not just `IOException`